File: loom-workshop-java/src/main/java/nl/theexperts/loomworkshopjava/api/PipelineController.java
package nl.theexperts.loomworkshopjava.api;

import nl.theexperts.loomworkshopjava.service.PipelineService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class PipelineController {

    private final PipelineService pipelineService;

    public PipelineController(PipelineService pipelineService) {
        this.pipelineService = pipelineService;
    }

    @PostMapping("/handle-simple-pipeline")
    public List<String> simplePipeline(@RequestBody List<String> list) {
       return pipelineService.processLargeDataSet(list);
    }
}

File: loom-workshop/src/main/kotlin/nl/theexperts/loomworkshop/api/PipelineController.kt
package nl.theexperts.loomworkshop.api

import nl.theexperts.loomworkshop.service.PipelineService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
    class PipelineController(val pipelineService: PipelineService) {
    //call service
    @PostMapping("/handle-simple-pipeline")
    fun simplePipeline(@RequestBody list: List<String> ) =
        pipelineService.processLargeDataSet(list)
}